### PR TITLE
Fix Symbol View In SyntaxTreeVizualizer

### LINKED ISSUE #504: Syntax Visualizer shows SyntaxNodes but no Symbols
In Visual Studio 2019 and Visual Studio 2019 Preview the Syntax Visualizer shows only the type and kind name of a corresponding Symbol - but the grid is empty.
This worked in Visual Studio 2017.

grafik

### PR DESCRIPTION
**Pull request by jmarolf**
fixes #504

## Before

image

## After

image

### CODE REVIEW

**Comment by sharwell** [comment]
Can you add screenshot(s) when this is ready to review?

File: tests/VisualStudio.Roslyn.SDK/Roslyn.SDK.IntegrationTests/InProcess/InProcComponent.cs [hunk 1]
```diff
@@ -30,25 +30,26 @@ protected async Task ExecuteCommandAsync(string commandName, string args = "")
...
             where TInterface : class
         {
             await JoinableTaskFactory.SwitchToMainThreadAsync();
 
-            var serviceProvider = (IAsyncServiceProvider2)await AsyncServiceProvider.GlobalProvider.GetServiceAsync(typeof(SAsyncServiceProvider));
+            var serviceProvider = (IAsyncServiceProvider2?)await AsyncServiceProvider.GlobalProvider.GetServiceAsync(typeof(SAsyncServiceProvider));
             Assumes.Present(serviceProvider);
-            return (TInterface)await serviceProvider.GetServiceAsync(typeof(TService));
+            var result = (TInterface?)await serviceProvider.GetServiceAsync(typeof(TService));
+            return result is null ? throw new Exception($"Unable to get service {typeof(TInterface).FullName}") : result;
```
Comment (sharwell) [inline tests/VisualStudio.Roslyn.SDK/Roslyn.SDK.IntegrationTests/InProcess/InProcComponent.cs:46]:
```suggestion
            return result ?? throw new Exception($"Unable to get service '{typeof(TInterface).FullName}'");
```

Or possibly:

```suggestion
            Assumes.Present(result);
            return result;
```

File: tests/VisualStudio.Roslyn.SDK/Roslyn.SDK.IntegrationTests/InProcess/InProcComponent.cs [hunk 2]
```diff
@@ -30,25 +30,26 @@ protected async Task ExecuteCommandAsync(string commandName, string args = "")
         {
             await JoinableTaskFactory.SwitchToMainThreadAsync();
 
-            var dte = await GetGlobalServiceAsync<SDTE, EnvDTE.DTE>();
+            var dte = await GetRequiredGlobalServiceAsync<SDTE, EnvDTE.DTE>();
             dte.ExecuteCommand(commandName, args);
         }
 
-        protected async Task<TInterface> GetGlobalServiceAsync<TService, TInterface>()
+        protected async Task<TInterface> GetRequiredGlobalServiceAsync<TService, TInterface>()
```
Comment (sharwell) [inline tests/VisualStudio.Roslyn.SDK/Roslyn.SDK.IntegrationTests/InProcess/InProcComponent.cs:37]: 💭 What if we keep the same name but have the new behavior?

File: src/VisualStudio.Roslyn.SDK/Roslyn.SDK/Properties/launchSettings.json [hunk 3]
```diff
@@ -1,9 +1,9 @@
 {
   "profiles": {
-    "Templates.VisualStudio.2017": {
+    "Templates.VisualStudio.2019": {
       "commandName": "Executable",
       "executablePath": "$(DevEnvDir)devenv.exe",
-      "commandLineArgs": "/rootsuffix RoslynDev /log"
+      "commandLineArgs": "/rootsuffix Roslyn /log"
```
Comment (sharwell) [inline src/VisualStudio.Roslyn.SDK/Roslyn.SDK/Properties/launchSettings.json:6]:
```suggestion
      "commandLineArgs": "/rootsuffix $(VSSDKTargetPlatformRegRootSuffix) /log"
```

File: src/VisualStudio.Roslyn.SDK/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Control/SymbolDisplay/BasePropertyGridAdapter.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel;
+
+namespace Roslyn.SyntaxVisualizer.Control.SymbolDisplay
+{
+    internal abstract class BasePropertyGridAdapter : ICustomTypeDescriptor
```
Comment (sharwell) [inline src/VisualStudio.Roslyn.SDK/SyntaxVisualizer/Roslyn.SyntaxVisualizer.Control/SymbolDisplay/BasePropertyGridAdapter.cs:8]: 💭 Yeah there are a bunch of cases where I want to use this...

**Comment by jmarolf** [comment]
> Can you add screenshot(s) when this is ready to review?

I've added some screenshots. I'll mark this PR as non-draft once I've done some testing of different scenarios. I assume this will crash VS in some situations today

File: src/VisualStudio.Roslyn.SDK/Roslyn.SDK/Properties/launchSettings.json (hunk 3 above)
Comment (jmarolf) [reply src/VisualStudio.Roslyn.SDK/Roslyn.SDK/Properties/launchSettings.json:6]: does launch settings inherit this variable?

File: tests/VisualStudio.Roslyn.SDK/Roslyn.SDK.IntegrationTests/InProcess/InProcComponent.cs (hunk 2 above)
Comment (jmarolf) [reply tests/VisualStudio.Roslyn.SDK/Roslyn.SDK.IntegrationTests/InProcess/InProcComponent.cs:37]: I am find reverting it. I typically use `Required` for things that throw but considering the usage I am also ok with reverting it.

File: src/VisualStudio.Roslyn.SDK/Roslyn.SDK/Properties/launchSettings.json (hunk 3 above)
Comment (sharwell) [reply src/VisualStudio.Roslyn.SDK/Roslyn.SDK/Properties/launchSettings.json]: It does. It's super handy.

**Comment by bernd5** [comment]
Any new on this?

**Comment by sharwell** [comment]
> Any news on this?

@jmarolf and I are seeing very different experiences prior to this change, so we want to make sure we understand the true root cause of cases where information is not showing up. We haven't had time to complete that investigation yet.

**Comment by bernd5** [comment]
Can I help you somehow? Could I get some beta?
(For me it is okay if it is a bit buggy - better than what we have today)

**Comment by bernd5** [comment]
Can I install / build it by myself?

**Comment by bernd5** [comment]
Sorry, but I need to ask again. Can I help you somehow?

**Comment by bernd5** [comment]
Could I get your draft, please? How can I build it / install in my visual Studio?

**Comment by bernd5** [comment]
It's really sad that nothing happens here ...

**Comment by bernd5** [comment]
Do you have any news on this? We work a lot with roslyn and need the visualizer to use it effectively.

**Comment by sharwell** [comment]
@bernd5 if you are able to explain why it was broken originally, it would be a big help.

**Comment by bernd5** [comment]
I would really like to help you and debug it - but if I try so I VS does not stop at the breakpoints...
I just opened the solution, select "Roslyn.SyntaxVisualizer.Extension".
How can I debug it correctly / which debug args? Do I need some additional environtment adjustments?
I have VS 16.8 P2 installed.

**Comment by sharwell** [comment]
> How can I debug it correctly / which debug args?

After you build, Test Explorer will show a set of integration tests. You can launch the IDE for Run or Debug via a "special" test from this menu:

image

**Comment by bernd5** [comment]
If I try those tests I get:
System.Runtime.InteropServices.COMException: "Ausnahme von HRESULT: 0x8971001E"

**Comment by jmarolf** [comment]
closing in favor of https://github.com/dotnet/roslyn-sdk/pull/634

**Pull request closed without merging**